### DISCUSSION TITLE
How to find PYTHON_HOME after setup-python action?

### DISCUSSION THREAD

**Discussion by lostmsu** [community/community#25115] [Actions]
<p>I am using setup-python action to test Python embedding scenario. So I load libpythonX.Xm.so in the test, but for it to work properly I must call Py_SetPythonHome with a path to Python installation. How can I find it? setup-python does not appear to be setting PYTHON_HOME variable.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/lostmsu">@lostmsu</a> ,</p>
<p>You can try using the command " <strong>type -p python</strong>" to find the installation directory of Python.</p>
<p>A simple example:</p>
<pre><code>jobs:
  test:
    name: Test
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix:
        os: [windows-latest, ubuntu-latest, macos-latest]
    steps:
      - name: Checkout
        uses: actions/checkout@v2
        with:
          fetch-depth: 2

      - name: Setup Python
        uses: actions/setup-python@v1
        with:
          python-version: '3.x'

      - name: Find installation directory of Python
        shell: bash
        run: |
          type -p python
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/lostmsu">@lostmsu</a> ,</p>
<p>How are things going? Have you tried my above suggestion that using the command " <strong>type -p python</strong>" to find the installation directory of Python? Is it helpful to you?</p>
<p>any progress, feel free to contact me. Thanks.</p>

**Comment by lostmsu** [comment, marked as answer]
<p><span class="mention">@brightranI</span> ended up using undocumented <strong>pythonLocation</strong> environment variable, that is set by the setup-python task.</p>
